core/observe.c

```diff
@@ -925,12 +927,11 @@ int lwm2m_observe(lwm2m_context_t * contextP,
     transactionP = transaction_new(clientP->sessionH, COAP_GET, clientP->altPath, uriP, contextP->nextMID++, 4, token);
     if (transactionP == NULL)
     {
+        observationP->clientP->observationList = (lwm2m_observation_t *)LWM2M_LIST_RM(observationP->clientP->observationList, observationP->id, NULL);
```

Comment:
I agree the existing registration may be removed. But in the first place you decided to modify it. So in case of failure, everything is removed. I
If the client sends a notification for a non-existing observation, the server replies with a RST message cancelling the observation.